frontend/src/js/preview-v2/Charts.tsx

```diff
@@ -39,16 +42,16 @@ type ChartProps = {
   statistics: PreviewStatistics[];
   className?: string;
   showPopup: (statistic: PreviewStatistics) => void;
+  page: number;
+  setPage: (page: number) => void;
 };
 
-export default function Charts({ statistics, className, showPopup }: ChartProps) {
-  const [index, setIndex] = useState<number>(0);
-
+export default function Charts({ statistics, className, showPopup, page, setPage }: ChartProps) {
   return (
     <>
       <Root className={className}>
         <DiagramContainer>
-          {statistics.slice(index * 4, (index + 1) * 4).map((statistic) => {
+          {statistics.slice(page * 4, (page + 1) * 4).map((statistic) => {
```

Comment:
maybe extract this to a memo hook for readablity.
e.g.
`const selectedPages = useMemo(() => statistics.slice(page * 4, (page + 1) * 4), [statistics, page])`